src/brainfuck/exceptions/BrainfuckException.java

```diff
@@ -0,0 +1,28 @@
+package brainfuck.exceptions;
+
+/**
+ * Superclass for all our application's exceptions.
+ * Extends RuntimeException so that all our exceptions are unchecked.
+ *
+ * @author Pierre-Emmanuel Novac
+ * @see <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/RuntimeException.html">RuntimeException</a>
+ */
+public abstract class BrainfuckException extends RuntimeException {
```

Comment:
The whole point is to `catch (BrainfuckException e)`there. It does not make sense to catch an `interface`. You can only catch an object instanciated from a subclass of `Throwable`.
As a side effect it allows us to declare all our exceptions as being unchecked since we extends `RuntimeException`.